src/main/java/hudson/scm/PerJobCredentialStore.java

```diff
@@ -105,8 +112,24 @@ private Object writeReplace() {
     private static final ThreadLocal<Boolean> IS_SAVING = new ThreadLocal<Boolean>();
 
     /*package*/ void migrateCredentials(SubversionSCM.DescriptorImpl descriptor) throws IOException {
-        for (Map.Entry<String, Credential> e : credentials.entrySet()) {
-            descriptor.migrateCredentials(project, e.getKey(), e.getValue());
+        Iterable<CredentialsStore> it = CredentialsProvider.lookupStores(project);
+        if (it != null && it.iterator().hasNext()) {
```

Comment:
@Vlatombe @recena 

FWIW I do not use Guava in Jenkins plugins unless there is a gun pointed at my head. It has caused more pain due to their differential view of backwards compatibility from that of the Jenkins project. I got badly burned in the 1.4xx versions of Jenkins due to Guava changes and had to rip out its usage. I recommend you avoid Guava in Jenkins plugins like one would avoid the plague